### DISCUSSION TITLE
Question, how will locutus distribute the frontend to the smart contract?

### DISCUSSION THREAD

**Discussion by Destroyinator69420** [freenet/freenet-core#305] [Q&A] [answered]
From what I understand about locutus, I understand that one of the goals is that locutus integrates with the browser, and that joining a smart contract should be like installing a browser extension, but how will the frontend for the smart contract make its way onto my computer? I understand that Zeronet, a similar idea used Bittorrent both as a CDN and a backend for decentralized apps written in Coffeescript, so how will Locutus download the Front end to my computer? Will it be similar to installing a Plugin in Freenet?

**Comment by sanity** [comment]
Hi, have you seen the introductory video? Here is a [link](https://youtu.be/d31jmv5Tx5k?t=1560) to the relevant part. Let me know if it doesn't answer your question.

**Comment by iduartgomez** [comment, marked as answer]
The short answer is that the frontend is the state of one contract and is accessed thought the node websocket api. Check the book for an example.
